### PR TITLE
tests: Bring latests CRIO tests.

### LINKED ISSUE #921: improve and/or document the way we run CRI-O tests
We currently run a subset of the CRI-O tests from `github.com/kubernetes-incubator/cri-o/tests`, but there is no documentation of what approach we follow. We also need to find a way to have a more reliable process, since our tests can be outdated at some point.

For reference, this was a previous discussion:
https://github.com/clearcontainers/tests/pull/919#pullrequestreview-97518645

### PR DESCRIPTION
**Pull request by GabyCT**
With this will retrieve the cri-o tests from the repository and we
will skip the tests that currently are not working.

Fixes #921

Signed-off-by: Gabriela Cervantes <gabriela.cervantes.tellez@intel.com>

### CODE REVIEW

**Comment by sboeuf** [comment]
@GabyCT very nice, this way we won't have to manually modify and maintain our own `crio.bats` file !
LGTM

File: integration/cri-o/cri-o.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+#
+# Copyright (c) 2017 Intel Corporation
```
Comment (chavafg) [inline integration/cri-o/cri-o.sh]: 2018?

File: integration/cri-o/cri-o.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+SCRIPT_PATH=$(dirname "$(readlink -f "$0")")
+source "${SCRIPT_PATH}/crio_skip_tests.sh"
+
+echo "Obtain CRI-O repository"
+go get -d github.com/kubernetes-incubator/cri-o || true
```
Comment (chavafg) [inline integration/cri-o/cri-o.sh]: after cloning the cri-o repo, can you checkout to the version that we support?  
something like: https://github.com/clearcontainers/tests/blob/master/.ci/install_crio.sh#L20-L28

File: integration/cri-o/crio_skip_tests.sh [hunk 3]
```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+#
+# Copyright (c) 2017 Intel Corporation
```
Comment (chavafg) [inline integration/cri-o/crio_skip_tests.sh]: 2018?

File: integration/cri-o/cri-o.sh [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+OLD_IFS=$IFS
+IFS=''
+
+# Skip CRI-O tests that currently are not working
+pushd $GOPATH/src/github.com/kubernetes-incubator/cri-o/test/
+for i in ${skipCRIOTests[@]}
+do
+	sed -i '/'${i}'/a skip \"This is not working\"' "$GOPATH/src/github.com/kubernetes-incubator/cri-o/test/ctr.bats"
```
Comment (jodh-intel) [inline integration/cri-o/cri-o.sh]: Neat! Is there a general issue number you could add to the skip message out of interest?

File: integration/cri-o/cri-o.sh (hunk 2 above)
Comment (grahamwhaley) [reply integration/cri-o/cri-o.sh]: you might want to add a `-u` to the `go get` as well, just in case the repo already exists etc.

**Comment by GabyCT** [comment]
@chavafg changes applied

File: integration/cri-o/cri-o.sh [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+SCRIPT_PATH=$(dirname "$(readlink -f "$0")")
+source "${SCRIPT_PATH}/crio_skip_tests.sh"
+
+crio_repository="github.com/kubernetes-incubator/cri-o"
+runtime_repository="github.com/clearcontainers/runtime"
+
+echo "Obtain CRI-O supported version"
+go get -u -d "${runtime_repository}" || true
+versions_file="$GOPATH/src/${runtime_repository}/versions.txt"
```
Comment (chavafg) [inline integration/cri-o/cri-o.sh]: sorry, forgot to mention that we already have a function for getting the versions:
```
source .ci/lib.sh
get_cc_versions
```

File: integration/cri-o/cri-o.sh [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+fi
+
+OLD_IFS=$IFS
+IFS=''
+
+# Skip CRI-O tests that currently are not working
+pushd $GOPATH/src/${crio_repository}/test/
+for i in ${skipCRIOTests[@]}
+do
+	sed -i '/'${i}'/a skip \"This is not working (Issue https://github.com/clearcontainers/tests/issues/943)\"' "$GOPATH/src/github.com/kubernetes-incubator/cri-o/test/ctr.bats"
```
Comment (devimc) [inline integration/cri-o/cri-o.sh]: nit: use ${crio_repository}

**Pull request merged**